scripts/actions/mobskills/frigid_shuffle.lua

```diff
@@ -8,11 +8,17 @@
 local mobskillObject = {}
 
 mobskillObject.onMobSkillCheck = function(target, mob, skill)
-    return 0
+    return xi.apkallu.canUseAbility(mob, 30)
 end
 
 mobskillObject.onMobWeaponSkill = function(target, mob, skill)
-    skill:setMsg(xi.mobskills.mobStatusEffectMove(mob, target, xi.effect.PARALYSIS, 50, 0, 60))
+    if not target:isFacing(mob) then
+        skill:setMsg(xi.msg.basic.SKILL_MISS)
+        return
+    end
+
+    local duration = xi.mobskills.calculateDuration(skill:getTP(), 60, 120)
```

Comment:
Is there a reason we are assuming duration always changes with varying TP?

Reply:
I think there are two other options:

1. Purely random
2. It can be resisted

Having duration scale with TP makes more sense design wise than randomness. None of the wikis mention this nor does Jimmayus' mob spreadsheet. I think it'd just require lots more retail testing which is a little hard because you can't know the mobs actual TP (unless you can charm them). Maybe monstrosity would be able to tell us?

Reply:
I don't think any durations are purely random other than the bind research tracent did. Would be the first time I am hearing about that. I thought it was just a resist check. I just have a feeling we are going to end up removing all of these at some point.

You would have to get it to TP above 50% (at 3000 tp) and see durations. Then get it below 25% for 1k tp and see if the durations are longer/shorter/same.

If the duration varies with TP it looks like it was noted during his research for monstrosity (see example) 
![image](https://github.com/user-attachments/assets/922c9731-a3af-4567-88c5-36e195cd5f51)

Reply:
I see. Perhaps I need to talk to Jimmayus about this then to better understand the duration range. I don't understand why he would provide a range if the lower bound is simply a resist value. There must be a reason though.
![image](https://github.com/user-attachments/assets/1f78dad7-0b11-4af9-a1a7-fe05d7906512)

Reply:
If its a half resist check it still lands but the duration is halved. I would follow up with him before you spend all this time doing the calc duration stuff and it has to be re-done. He might have an answer for you immediately

Reply:
After talking to Jimmayus I believe the proper solution here is to use `math.random`. I want to do some thorough testing of at least some mobskill debuff durations and revise all places I've used the TP to calculate the duration but that is outside the scope of this PR.

Reply:
lgtm!